angular-rails-app/app/assets/javascripts/property/property.html

```diff
@@ -131,6 +134,15 @@ <h4 class="table-title">Open HPD Violations</h4>
                 <td>{{ v.certified_date }}</td>
               </tr>
             </table>
+            <div class="text-center">
+                <dir-pagination-controls
+                pagination-id="hpd_violations"
+                on-page-change="vm.hpdViolationsPageChanged(newPageNumber)"
+                max-size="10"
+                direction-links="true"
+                boundary-links="true" >
+              </dir-pagination-controls>
+            </div>
```

Comment:
There's a lot of repetition here: let's make a "paged property related data" component and reuse it. I think you can inject the specific bindings (td elements) via "transclusion". The other parameters can just be provided via the isolate scope.